src/containers/Sidebar/SidebarNodeView.tsx

```diff
@@ -3,65 +3,56 @@ import { connect } from 'react-redux';
 import { Button } from 'antd';
 import { bindActionCreators, Dispatch } from 'redux';
 import { selectThreeD, ThreeDState } from '../../modules/threed';
-import {
-  ExtendedAsset,
-  fetchAsset,
-  selectAssets,
-  AssetsState,
-} from '../../modules/assets';
+import { fetchAsset, selectAssets, AssetsState } from '../../modules/assets';
 import { RootState } from '../../reducers/index';
 import { sdk } from '../../index';
 import MapModelToAssetForm from '../MapModelToAssetForm';
 import MapNodeToAssetForm from '../MapNodeToAssetForm';
+import {
+  resetAppState,
+  AppState,
+  selectApp,
+  setModelAndRevisionAndNode,
+} from '../../modules/app';
 
-type OrigProps = {
-  asset: ExtendedAsset;
-  modelId: number;
-  revisionId: number;
-  nodeId: number;
-  onNodeIdChange: (nodeId?: number) => void;
-  onAssetIdChange: (id?: number) => void;
-};
+type OrigProps = {};
 
 type Props = {
+  app: AppState;
   assets: AssetsState;
   threed: ThreeDState;
+  setModelAndRevisionAndNode: typeof setModelAndRevisionAndNode;
+  resetAppState: typeof resetAppState;
 } & OrigProps;
 
 type State = {};
 
 class NodeDrawer extends React.Component<Props, State> {
   readonly state: Readonly<State> = {};
 
-  get rootAssetLinkedToRevision() {
-    const { modelId, revisionId } = this.props;
-    const { representsAsset } = this.props.threed;
-    return Number(
-      Object.keys(representsAsset).find(
-        assetId =>
-          representsAsset[Number(assetId)].modelId === modelId &&
-          representsAsset[Number(assetId)].revisionId === revisionId
-      )
-    );
-  }
-
   selectParentClicked = async () => {
-    const { modelId, revisionId, nodeId } = this.props;
-    const parent = await sdk.revisions3D.list3DNodes(modelId, revisionId, {
+    const { modelId, revisionId, nodeId, rootAssetId } = this.props.app;
+    const parent = await sdk.revisions3D.list3DNodes(modelId!, revisionId!, {
       nodeId,
     });
-    this.props.onNodeIdChange(parent.items[0].parentId);
+    this.props.setModelAndRevisionAndNode(
+      rootAssetId!,
+      parent.items[0].parentId
```

Comment:
Does this always exist? Or can we have a zero length array?

Reply:
added check